src/aw-client.ts

```diff
@@ -75,88 +75,95 @@ export class AWClient {
+                buckets[bucket].last_updated = new Date(buckets[bucket].last_updated);
             }
-          });
-          return buckets;
         });
+        return buckets;
     }
 
     public async getBucketInfo(bucketId: string): Promise<IBucket> {
-        return this.req.get("/0/buckets/" + bucketId)
-        .then(res => res.data)
-        .then(bucket => {
-          bucket.created = new Date(bucket.created);
-          return bucket;
-        });
+        let bucket = (await this.req.get(`/0/buckets/${bucketId}`)).data;
+        bucket.created = new Date(bucket.created);
+        return bucket;
     }
 
-    public getEvents(bucketId: string, params: { [k: string]: any }): Promise<IEvent[]> {
-        return this.req.get("/0/buckets/" + bucketId + "/events", { params }).then(res => res.data)
-        .then(events => {
-          events.forEach((event: IEvent) => {
+    public async getEvents(bucketId: string, params: { [k: string]: any }): Promise<IEvent[]> {
+        let events = (await this.req.get("/0/buckets/" + bucketId + "/events", { params })).data;
+        events.forEach((event: IEvent) => {
             event.timestamp = new Date(event.timestamp);
-          });
-          return events;
         });
+        return events;
     }
 
-    public countEvents(bucketId: string, startTime?: Date, endTime?: Date) {
+    public async countEvents(bucketId: string, startTime?: Date, endTime?: Date) {
         const params = {
             starttime: startTime ? startTime.toISOString() : null,
             endtime: endTime ? endTime.toISOString() : null,
         };
         return this.req.get("/0/buckets/" + bucketId + "/events/count", { params });
     }
 
-    public insertEvent(bucketId: string, event: IEvent) {
+    public async insertEvent(bucketId: string, event: IEvent): Promise<IEvent> {
         return this.insertEvents(bucketId, [event]).then(events => events[0]);
     }
 
-    public insertEvents(bucketId: string, events: IEvent[]): Promise<IEvent[]> {
-        return this.req.post("/0/buckets/" + bucketId + "/events", events)
-        .then(res => res.data)
-        .then(insertedEvents => {
-          if (!Array.isArray(insertedEvents)) {
+    public async insertEvents(bucketId: string, events: IEvent[]): Promise<IEvent[]> {
+        let insertedEvents = (await this.req.post("/0/buckets/" + bucketId + "/events", events)).data;
+        if (!Array.isArray(insertedEvents)) {
             insertedEvents = [insertedEvents];
-          }
-          insertedEvents.forEach((event: IEvent) => {
+        }
+        insertedEvents.forEach((event: IEvent) => {
             event.timestamp = new Date(event.timestamp);
-          });
-          return insertedEvents;
         });
+        return insertedEvents;
+    }
+
+    // Just an alias for insertEvent requiring the event to have an ID assigned
+    public async replaceEvent(bucketId: string, event: IEvent): Promise<IEvent> {
```

Comment:
Does this work? I've never tried.

Reply:
Apparently it does, at first I used `deleteEvent` followed by `insertEvent` but that somehow didn't leave any event at all (surprisingly). This worked fine though, might want to add a test to ensure it keeps working though.

Reply:
It should work since the primary key in the datastore is replaced.
Add a unittest in aw-core and then we can assume that this is expected behavior, should be pretty useful.

Reply:
Not going to do that right now, let's just keep it in mind.